### PR TITLE
Phani | BAH-3080 | Update Appointment summary page

### PR DESCRIPTION
**Pull request by Phanindra-tw**
This existing Appointments summary page is written in Angular. This PR updates the page with the equivalent React components.
This PR adds the provision to view summary grouped by Specialities, Providers and Location. By default only the table consisting of appointments grouped by services will be shown. Showing the other tables is driven by config variable `enableDetailedSummaryView` When this variable is set to true it shows all the 4 tables ( Summary based on Specialities, Providers, Services, Locations ), if this variable is not passed or set to false in config, it would show only 1 table(Summary based on Services)
JIRA Ref: [Appointment Summary](https://bahmni.atlassian.net/jira/software/c/projects/BAH/boards/55?modal=detail&selectedIssue=BAH-3080)

Screenshots:

When the configuration is enabled:
Screenshot 2023-06-28 at 12 15 49 PM
Screenshot 2023-06-28 at 12 16 21 PM

When the configuration variable is disabled/not passed:
Screenshot 2023-12-14 at 10 19 08 AM

### CODE REVIEW

File: i18n/appointments/locale_en.json [hunk 1]
```diff
@@ -130,6 +130,7 @@
   "PATIENT_LIST_VIEW_SEARCH_HEADER": "Search",
   "NO_APPOINTMENT_LIST_VIEW": "No appointments found",
   "NO_APPOINTMENT_CALENDAR_VIEW": "No appointments found",
+  "NO_APPOINTMENT_SUMMARY_VIEW": "This week, no appointments are available for any of the {gridName}",
```
Comment (gsluthra) [inline i18n/appointments/locale_en.json:133]: What does `{gridName}` mean? During translation, the translators won't understand this word. Can you rephrase the variable to something meaningful for translators to understand. Is this `daysOrWeeks`? or `serviceName`?

Comment (Phanindra-tw) [reply i18n/appointments/locale_en.json:133]: Hey @gsluthra, The gridName is a param which is being passed from the GridSummary Component. https://github.com/Bahmni/openmrs-module-appointments-frontend/pull/291/files#diff-de2a9964b0394faa6785c534604c44ddb74e68bfddcb4e0e317ca850cc9f0c94R108
`{intl.formatMessage({id: "NO_APPOINTMENT_SUMMARY_VIEW"}, {gridName: gridName})}`

 The param gridName is already translated and sent to the GridSummary so as is would be replaced.
`const specialitiesTitle = <FormattedMessage id={'SPECIALITIES'} defaultMessage={'Specialities'}/>`
`<GridSummary gridData={locationData} weekStartDate={startDate} onClick={goToListView} gridName={locationsTitle}/>`

Comment (gsluthra) [reply i18n/appointments/locale_en.json:133]: Then this should be: `This week, no appointments are available for any of the {locationName}`  
The reason I say this is because depending on the context of the word in { }, the translators will decide how to write this message. If its locationName, serviceName, Dates, etc... the sentence structure might change.

**Comment by gsluthra** [comment]
For the rest of the code, I don't have any comments. Seems fine to me (I am no react expert).

File: i18n/appointments/locale_en.json (hunk 1 above)
Comment (abinaya-u) [reply i18n/appointments/locale_en.json:133]: @gsluthra GridSummary is a generic component and we are accepting gridName and gridData as props, so the gridName is already translated based on the locale and we have also added a default message.

Comment (gsluthra) [reply i18n/appointments/locale_en.json:133]: I am not convinced. Can you please share example of what final messages might looks like, so i can decide if this makes sense. Language translators need to know some context of the message, and gridName is technical and doesn't tell people what the sentence eventually is going to say.

Comment (gsluthra) [reply i18n/appointments/locale_en.json:133]: Please see section titled "Word Order" in this article to understand what I mean: https://ux.shopify.com/how-to-concatenate-localized-strings-mindfully-bcc7b6d4961a

Comment (Phanindra-tw) [reply i18n/appointments/locale_en.json:133]: Hi Gurpreet, Based on the gridName the message would be like
Incase of Specialities :
This week, no appointments are available for any of the Specialities
For Providers:
This week, no appointments are available for any of the Providers

Screenshot for Reference:
Screenshot 2023-12-14 at 10 09 01 AM

Comment (gsluthra) [reply i18n/appointments/locale_en.json:133]: In that case the variable can be renamed to `{specialties_or_providers}`

File: ui/i18n/appointments/locale_en.json [hunk 2]
```diff
@@ -229,5 +229,8 @@
   "CHANGE_DATE_TO_LABEL": "Change ${moment(originalAppointmentDate).format('Do MMM')} to",
   "PATIENT_DOUBLE_BOOKING_CONFLICT_MESSAGE": "Current {currentAppointmentService} request conflicts with {existingAppointmentService} appointment on and",
   "NO_SERVICE_CONFLICTS_DEFAULT_TEXT": "The {label} service you had selected for the appointment(s) is not available during below listed dates",
-  "PROVIDER_ERROR_MESSAGE": "Please select only a maximum of {maxAppointmentProviders} provider(s)"
+  "PROVIDER_ERROR_MESSAGE": "Please select only a maximum of {maxAppointmentProviders} provider(s)",
+  "PROVIDERS": "Providers",
```
Comment (gsluthra) [inline ui/i18n/appointments/locale_en.json:233]: These keynames are the same as the key names in the file above. What is the difference in both. Is one "headings"? I feel the Key name should be more specific where this is being used.

File: i18n/appointments/locale_en.json (hunk 1 above)
Comment (gsluthra) [reply i18n/appointments/locale_en.json]: The locale strings have been split into individual different strings now by @Phanindra-tw . looks good. thanks.

**Review by gsluthra — APPROVED** [review]
